layouts/partials/settings.html

```diff
@@ -1,6 +1,18 @@
 {{- range $i, $def := (getJSON (printf "themes/tvpage/defaults/%s.json" .Type)).option -}}
 {{- $.Scratch.SetInMap "settings" .code .value -}}
 {{- end -}}
+{{- $type := .Type -}}
```

Comment:
Please move `{{- $type := .Type -}}` at the top so you can reuse it on:
`:= (getJSON (printf "themes/tvpage/defaults/%s.json" $type)).option -}}`